oteapi/models/genericconfig.py

```diff
@@ -17,34 +17,31 @@ def __contains__(self, name: "Any") -> bool:
         return self.__dict__.__contains__(name)
 
     def __delitem__(self, key: str) -> None:
-        """Enable deletion access through subscription."""
-        if key in dir(self):
-            self.__delattr__(key)
+        """Enable deletion access through subscription.
+
+        If the item is a pydantic field, reset it and remove it from the set of set
+        fields. Otherwise, delete the attribute.
+
+        """
+        if key in self.__dict__:
             if key in self.__fields__:
-                del self.__fields__[key]
+                # Reset field to default and remove from set of set fields
+                setattr(self, key, self.__fields__[key].default)
                 self.__fields_set__.remove(key)  # pylint: disable=no-member
+            else:
+                self.__delattr__(key)
         else:
             raise KeyError(key)
 
     def __getitem__(self, key: str) -> "Any":
         """Enable read access through subscription."""
-        if key in dir(self):
+        if key in self.__dict__:
             return getattr(self, key)
         raise KeyError(key)
 
-    def __setattr__(self, name: str, value: "Any") -> None:
```

Comment:
Because it's already implemented in the way we want from the side of pydantic.
One can already do: `my_attr_dict.some_key = some_value` and get type validation should the `some_key` attribute/field/key be a pydantic `Field` defined in some model/class that sub-classes `AttrDict` (like the `SessionUpdate` and other classes).

Reply:
I think the original intent of raising if the type does not match is obstructive. I'd rather have the model cast the value to the type. And note this will again only be an "issue" if the set attribute/field/key is one defined as a pydantic model `Field`.